File: service/src/main/java/com/testing/edu/service/StatisticService.java
package com.testing.edu.service;


import com.testing.edu.entity.User;

public interface StatisticService {

    /**
     * Count Subjects
     * @return amount of subjects
     */
    Long countSubjects();

    /**
     * Count Subjects by Lecturer id
     * @return amount of subjects
     */
    Long countSubjectsByLecturerId(Long lecturerId);

    /**
     * Count Lecturers
     * @return amount of lecturers
     */
    Long countLecturers();

    /**
     * Count Groups
     * @return amount of groups
     */
    Long countGroups();

    /**
     * Count Students
     * @return amount of Students
     */
    Long countStudents();

    /**
     * Count test witch have lecturer
     * @return
     */
    Long countTestByLecturerId(Long lecturerId);

    /**
     * Count results witch have lecturer
     * @return
     */
    Long countResultsByLecturerId(Long lecturerId);

    /**
     * find user with username
     * @param username
     * @return
     */
    User employeeExist(String username);
}

File: web/src/main/webapp/resources/app/admin/services/StatisticService.js
angular
    .module('adminModule')
    .factory('StatisticService', function ($http) {
        var getData = function (type) {
            return $http.get('/admin/statistics/' + type)
                .then(function(result) {
                    return result.data;
                });
        };
        return {
            subjects: function() { return getData('subjects'); },
            lecturers: function() { return getData('lecturers'); },
            groups: function() { return getData('groups'); },
            students: function() {return getData('students'); },
            results: function() {return getData('results'); },
            employee: function() { return getData('employee'); }
        }
    });
